src/kontrol/deployment.py

```diff
@@ -0,0 +1,123 @@
+class DeploymentSummary:
+    SOLIDITY_VERSION = '^0.8.13'
+
+    name: str
+    commands: list[str]
+    accounts: dict[str, str]  # address, name
+    code: dict[str, str]
+
+    def __init__(self, name: str, accounts: dict | None = None) -> None:
+        self.commands = []
+        self.accounts = accounts if accounts is not None else {}
+        self.name = name
+        self.code = {}
+        for acc_key in list(self.accounts):
+            self.accounts[acc_key] = self.accounts[acc_key]
+
+    def generate_header(self) -> list[str]:
+        lines = []
+        lines.append(f'pragma solidity {self.SOLIDITY_VERSION};\n')
+        lines.append('import { Vm } from "forge-std/Vm.sol";\n')
+        return lines
+
+    def generate_code_contract(self) -> list[str]:
+        lines = []
+        lines.append(f'contract {self.name}Code ' + '{')
+        for code_alias, code in self.code.items():
+            lines.append(f'\tbytes constant internal {code_alias}Code = hex{code!r};')
+        lines.append('}')
+        return lines
+
+    def generate_main_contract(self) -> list[str]:
+        lines = []
+        lines.append('')
+
+        lines.append(f'contract {self.name} is {self.name}Code ' + '{')
+        # Appending the cheatcode address to be able to avoid extending `Test`
+        lines.append('\t// Cheat code address, 0x7109709ECfa91a80626fF3989D68f67F5b1DD12D')
+        lines.append(
+            '\taddress internal constant VM_ADDRESS = address(uint160(uint256(keccak256("hevm cheat code"))));'
+        )
+        lines.append('\tVm internal constant vm = Vm(VM_ADDRESS);\n')
+
+        for acc_key in list(self.accounts):
+            lines.append('\taddress internal constant ' + self.accounts[acc_key] + 'Address = ' + acc_key + ';')
+
+        lines.append('\n')
+
+        lines.append('\tfunction recreateDeployment() public {')
+
+        lines.append('\t\tbytes32 slot;')
+        lines.append('\t\tbytes32 value;')
+
+        for command in self.commands:
+            lines.append('\t\t' + command + ';')
+
+        lines.append('\t}')
+
+        lines.append('}')
+        return lines
+
+    def generate_condensed_file(self) -> list[str]:
+        lines = self.generate_header()
+        lines += self.generate_code_contract()
+        lines += self.generate_main_contract()
+        return lines
+
+    def generate_main_contract_file(self) -> list[str]:
+        lines = self.generate_header()
+        lines.append('import { ' + self.name + 'Code } from "./' + self.name + 'Code.sol";')
+        lines += self.generate_main_contract()
+        return lines
+
+    def generate_code_contract_file(self) -> list[str]:
+        lines = self.generate_header()
```

Comment:
This will add `import { Vm } from "forge-std/Vm.sol";` to the code contract, but it's actually not needed. Better remove it

Reply:
Fixed.